### PR TITLE
Fixes for Observable.controlled along with more unit tests

### PR DESCRIPTION
**Pull request by kcarl-adb**

### CODE REVIEW

File: src/core/backpressure/controlled.js [hunk 1]
```diff
@@ -57,25 +58,23 @@
...
       onNext: function (value) {
-        var hasRequested = false;
-
-        if (this.requestedCount === 0) {
+        if (this.requestedCount <= 0) {
           this.enableQueue && this.queue.push(Notification.createOnNext(value));
         } else {
-          (this.requestedCount !== -1 && this.requestedCount-- === 0) && this.disposeCurrentRequest();
-          hasRequested = true;
+          (this.requestedCount-- === 0) && this.disposeCurrentRequest();
```
Comment (elhesuu) [inline src/core/backpressure/controlled.js]: Assignment in a conditional?

**Comment by mattpodwysocki** [comment]
@kcarl-adb do we want to throw on invalid values or just set it to 1?

**Comment by mattpodwysocki** [comment]
@kcarl-adb also, please sign the [CLA](https://cla2.msopentech.com/)

**Comment by paulpdaniels** [comment]
Looks good.

Missing [semicolon](https://github.com/kcarl-adb/RxJS/blame/controlled/src/core/backpressure/controlled.js#L98)

**Comment by mattpodwysocki** [comment]
@paulpdaniels I'll pose the question to you as well. What should we do on invalid value sent to request?

**Comment by paulpdaniels** [comment]
My vote is on throw.  I can't think of a scenario where something good could come from letting people throw whatever at `request()`.  At best they are doing it intentionally and then an explicit conversion would be a kind of documentation for why they are doing it that way. At worst they made a type error somewhere in which case we would would just compound the debugging process by creating unexpected behavior.

**Comment by kcarl-adb** [comment]
Requesting permission for CLA but not able to right now.

**Pull request closed without merging**